octobot_trading/personal_data/orders/order.py

```diff
@@ -477,41 +477,30 @@ def is_short(self):
-                if self.taker_or_maker is None:
-                    self._update_taker_maker()
-            except KeyError:
-                logging.get_logger(self.__class__.__name__).warning("Failed to parse order side and type")
-
-        price = raw_order.get(enums.ExchangeConstantsOrderColumns.PRICE.value, 0.0) or 0.0
-        # TODO replace with := when cython will be supporting it
-        stop_price = raw_order.get(enums.ExchangeConstantsOrderColumns.STOP_PRICE.value, None)
-        if stop_price is not None and (price is None or price == 0):
-            # parse stop price when available
-            price = stop_price
-        filled_price = decimal.Decimal(str(price))
-        # set average price with real average price if available, use filled_price otherwise
-        average_price = decimal.Decimal(str(raw_order.get(enums.ExchangeConstantsOrderColumns.AVERAGE.value, 0.0)
-                                            or filled_price))
-
-        return self.update(
-            symbol=str(raw_order.get(enums.ExchangeConstantsOrderColumns.SYMBOL.value, None)),
-            current_price=decimal.Decimal(str(price)),
-            quantity=decimal.Decimal(str(raw_order.get(enums.ExchangeConstantsOrderColumns.AMOUNT.value, 0.0) or 0.0)),
-            price=decimal.Decimal(str(price)),
-            status=order_util.parse_order_status(raw_order),
-            order_id=str(raw_order.get(enums.ExchangeConstantsOrderColumns.ID.value, None)),
-            quantity_filled=decimal.Decimal(str(raw_order.get(enums.ExchangeConstantsOrderColumns.FILLED.value, 0.0)
-                                                or 0.0)),
-            filled_price=decimal.Decimal(str(filled_price)),
-            average_price=decimal.Decimal(str(average_price)),
-            total_cost=decimal.Decimal(str(raw_order.get(enums.ExchangeConstantsOrderColumns.COST.value, 0.0) or 0.0)),
-            fee=order_util.parse_raw_fees(raw_order.get(enums.ExchangeConstantsOrderColumns.FEE.value, None)),
-            timestamp=raw_order.get(enums.ExchangeConstantsOrderColumns.TIMESTAMP.value, None),
-            reduce_only=raw_order.get(enums.ExchangeConstantsOrderColumns.REDUCE_ONLY.value, False)
-        )
+            if self.taker_or_maker is None:
+                self.taker_or_maker = raw_order[enums.ExchangeConstantsOrderColumns.TAKERORMAKER.value]
+            price = raw_order[enums.ExchangeConstantsOrderColumns.PRICE.value]
+            return self.update(
+                order_id=raw_order[enums.ExchangeConstantsOrderColumns.ID.value],
+                symbol=raw_order[enums.ExchangeConstantsOrderColumns.SYMBOL.value],
+                current_price=price,
+                quantity=raw_order[enums.ExchangeConstantsOrderColumns.AMOUNT.value],
+                price=price,
+                status=order_util.parse_order_status(raw_order),
+                quantity_filled=raw_order.get(enums.ExchangeConstantsOrderColumns.FILLED.value),
```

Comment:
we should use `[]` instead of `.get` if we are sure that when we are here we always have passed through parsers

Reply:
I'll change that for quantity_filled as we get that always anyway.

1. Should I also always set optional values like reduce_only and tag to None if they are missing?

2. I'm not sure what to set to the following when orders are unfilled:
quantity_filled - now its set to 0
filled_price - should it be None?
average_price - should it be None?

3. whats the reason for current price? Should it be the current price when the order was created?

4. is total_cost important? if yes should we always calculate cost? or just on filled orders? Should it include fees?